Cyborgs, MMIs, and other fixes

phil235 opened the pull request:
- Fix mmi deafened indefinitely. Fixes #9078
- Fix carp knocking borg. Fixes #7394
- Replace carp knockdown by stamina dmg (for human only) (now 5 bites instead of 7 to go from 100% to crit w/o armor). Fixes #7905
- Nerfs uranium wall radiation: 6 times less radiation (just double the effect of uranium floor), chain reaction range down to 1. Fixes #6841
- Fixes putting braindead/notincorpse/ghosted mmi in borgs. Putting a brain into a MMI sends a message to the ghost. Examining a MMI tells you if its brain is active. Fixes #4841. Fixes #7447
- Fixes brains taking eye damage and becoming blind. Fixes #6039

phil235 committed:
Fix mmi deafened indefinitely. 9078
Fix carp knocking borg. 7394
Replace carp knockdown by stamina dmg (for human only) (now 5 bites instead of 7 to go from 100% to crit w/o armor) 7905
Fix being stunned as a borg causing temporary blindness, deafness and muteness. 9061
Change robot out of power: now only stunned, can talk but not move or interact. Adding feedback in examine. 6170
Robots now take damage in the hottest of inferno (35,000+ K)  8840
Nerfs uranium wall radiation: 6 times less radiation (just double the effect of urafloor), chain reaction range down to 1. 6841
Fixes putting braindead/notincorpse/ghosted brains in mmi, and same for mmi in borgs, the action fails and sends a message to the ghost. 4841 7447
Fixes brains taking eye damage and becoming blind. 6039

phil235 committed:
adding changelog

Boggart commented:
I'm not real sure on the being unable to move with no power for borgs, in the (distant) past you could get to a charger and use it. On the other hand, you can now actually explain what's wrong and can call for help. (Actually fixing the chargers to work for them would be a huge PITA though.)

Overal though :+1:.

xxalpha commented:
Really really disagree with not being able to move as a cyborg when the power runs out. If that shit happens nowhere near a hallway, you can kiss your round goodbye. Being able to move is crucial to get back into the round.

No, screaming for help will not be better, nobody cares about others in this game.

Lumipharon commented:
The thing is, unless you suddenly get EMP'd to shit, you have pleny of time BEFORE you run out of power to do something about it.

Borg without power/cell running around silently always struck me as pretty silly.

I don't lean terribly in either way, but I thought a big part of borgs bein mute when stunned was that it was essentially impossible to emag a borg without it screaming over comms that you're a traitor, unless you're the roboticist.

phil235 committed:
Changing cyborg unconsciousness to be more like the other mobs. Cyborgs out of power are now able to move, hear, talk and interact but their speed is decreased. You can put inactive brains into MMIs again but not inactive MMIs into borg shells. Examining an MMI tells you if its brain is active.

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into BundleFix5

Conflicts:
	code/modules/mob/living/carbon/life.dm

phil235 commented:
Okay, so I changed the powerless borg to have pretty much no maluses, you can talk, hear, move and interact but not use your modules and you move slowly at a walking pace.
I changed brain to MMI to work even if the brain is braindead (mostly to avoid spamming the ghost with alert messages), but MMI to borg shell still won't work if the brain is not active.

Durkel commented:
Muh uranium walls.

Does this lower the effective range that they irradiate at as well? Because they only reach 3 tiles away from their source, making them actually very easy to avoid.

Jordie0608 commented:
They still irradiate mobs at the same 3 tile range.

GunHog commented on code/modules/mob/living/silicon/ai/ai.dm:
```diff
@@ -228,9 +228,16 @@ var/list/ai_list = list()
 			var/area/borg_area
 			for(var/mob/living/silicon/robot/R in connected_robots)
 				borg_area = get_area(R)
+				var/robot_status = "Normal"
```
While you are at it, this is a good chance to take RR's suggestion of using "Nominal" instead of "Normal" here!

RemieRichards commented on code/modules/mob/living/silicon/ai/ai.dm:
```diff
@@ -228,9 +228,16 @@ var/list/ai_list = list()
+					robot_status = "OFFLINE"
+				else if(!R.cell)
+					robot_status = "DEPOWERED"
+				else if(R.cell.charge <= 0)
```
You can combine this check with the above check
```else if(!R.cell || R.cell.charge <= 0)```
Since the right handside will only ever be evaluated if the cell exists.

phil235 replied:
will fix

palpatine213 commented:
Wait. So are EMPS the ONLY way to silence a borg now?

phil235 committed:
Readjusted rad effect from uranium walls.
mob/living/silicon/ai/ai.dm : one simplification plus "Normal" -> "Nominal".

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into BundleFix5

Conflicts:
	code/game/machinery/computer/ai_core.dm
	code/game/objects/items.dm
	code/game/objects/items/robot/robot_parts.dm
	code/modules/mob/living/carbon/brain/MMI.dm

phil235 commented:
@palpatine213 Pretty much yes. This should be good to go.

phil235 commented:
If my changes to robot stun and unconsciousness are too controversial I can remove it.

GunHog commented:
The only two things I oppose here are:
1. Borgs taking damage in a fire; Bad for Engiborgs and rogue AI plasma floods. I am guessing you are not going to budge on this one.
2. Depowered borgs being able to interact. That is quite an important ability on its own, and you expect a cell-less borg to be completely harmless in this state. The borg charger itself should be the only exception. If that is too snowflake, then I will leave the choice up to you.

xxalpha commented:
I agree with what GunHog said. I don't like cyborgs taking fire damage and they really shouldn't be able to interact if they have no power. Moving and being mute (how it is currently) works with the exception of the recharger problem.

phil235 commented:
Alright, I'll remove the cyborgs changes.

KorPhaeron commented:
Forcing the AI to make a choice between plasma fires and cyborgs killing everyone isn't a bad thing at all.

I'd rather a flash silence a borg than an EMP, as EMPs are unavoidable AoE attacks and zero fun to deal with as a cyborg. It's like being parapenned for the crime of being within viewrange of an antag

xxalpha commented:
That's the dumbest thing ever said, Kor.

KorPhaeron commented:
Why?

KorPhaeron commented:
To clarify further

Flash: Requires melee range/clicking the target, is clearly visible/very distinct in hand, unusable while stunned

EMP: Can be used while stunned (wizard, implant, changeling shriek), is undetectable, hits automatically within viewrange, drains your battery, does large amounts of damage

If people insist on there being a way to silence borgs, I'd rather it be the melee weapon than the unavoidable, undetectable, unstoppable attack.

If you're talking about fires, I dunno what to say. Plasma fires already kill almost the whole crew by themselves, they're the most powerful thing in the game.

Dorsisdwarf commented:
EMPs shouldn't mute borgs, but blinding them is better, so you can EMP a borg and it can't see who emped it

Boggart commented:
They used to get a static overlay when emped that partially blocked vision.

xxalpha commented:
Sorry I must have misread what you said, that makes sense. However I don't want even flashes silencing borgs.

phil235 committed:
Removes my changes to cyborgs.

phil235 commented:
I removed the controversial bits, the PR is ready to be merged.

Lumipharon commented:
It's really shit if there's no way to silence a borg - if you're trying to do anything sneaky and a borg sees you, you're basically fucked because you can't stop it immediately reporting you.

Dorsisdwarf commented:
Cult are the ONLY antags that can silence borgs, lumi.
